alvr/client_core/src/graphics/lobby.rs

```diff
@@ -310,11 +318,30 @@ impl LobbyRenderer {
         for section_glyph in section_glyphs {
             if let Some(outlined) = scaled_font.outline_glyph(section_glyph.glyph) {
                 let bounds = outlined.px_bounds();
+
                 outlined.draw(|x, y, alpha| {
-                    let x = x as usize + bounds.min.x as usize;
-                    let y = y as usize + bounds.min.y as usize;
-                    if x < HUD_TEXTURE_SIDE && y < HUD_TEXTURE_SIDE {
-                        buffer[(y * HUD_TEXTURE_SIDE + x) * 4 + 3] = (alpha * 255.0) as u8;
+                    let x = x as i32 + bounds.min.x as i32;
+                    let y = y as i32 + bounds.min.y as i32;
+
+                    if x >= MAX_BORDER_OFFSET
+                        && y >= MAX_BORDER_OFFSET
+                        && x < HUD_TEXTURE_SIDE as i32 - MAX_BORDER_OFFSET
+                        && y < HUD_TEXTURE_SIDE as i32 - MAX_BORDER_OFFSET
+                    {
```

Comment:
I don't understand how this works, so I'm just gonna trust you that it's correct and that it can't really be simplified ;)

Reply:
I don't think it can be simplified. I you saw the commit history, the border code used to be longer but also make more sense. Here i came up with a fast algorithm to create the border with just 8 assignments per pixel for arbitrary border sizes (probably if the border is greater than 5 it would look bad but yeah)